### ISSUE TITLE
dirPagination: on-page-change called 6 times each time I change page (version 0.9.4)

### ISSUE THREAD

**Issue by hplloyd** [michaelbromley/angularUtils#324]
Hi, thanks for contributing!

This project is maintained in my spare time, so in order to help me address your issue as quickly as 
possible, please provide as much of the following information as you can:

- Title: Please indicate the module in question, e.g. "dirPagination: X is broken when Y"
- If reporting on dirPagination, please include the version you are using (can be found in the package.json / bower.json file)
- If you are able to reproduce your issue on Plunker, this will vastly increase the chances of a rapid response from me.

-- Michael 

=======
(Delete the above. Fill in the rest as applicable)

**Description of issue**:

**Steps to reproduce**:

**Expected result**:

**Actual result**:

**Demo**: (for dirPagination, fork and modify this Plunk: http://plnkr.co/edit/b37IdFFJUokaeSummETX?p=preview)

Any relevant code:
```

```

**Comment by michaelbromley** [comment]
Hi,

Please could you [fork and edit this plunker demo](http://plnkr.co/edit/b37IdFFJUokaeSummETX?p=preview) and try to reproduce your issue? Paste the link to your updated demo here so I can take a look.

Note that the latest version is 0.11.0 - although I'm not sure if that should make a difference.

**Comment by hplloyd** [comment]
I have recreated on plunker... here is the link

[http://plnkr.co/edit/wMSxBaJgmQ94QoOEySwz]

I know what the issue is now.... but I dont know how to fix it

I like to have 2 pagination controls, one above and one below the list, which is fine as I use the pagination-id to keep everything in sync. Because each pagination control has an on-page-change attribute, you will see that each time I change the page the event is fired twice which in my real life case, causes to hits to my API (I am using server paging in my real life case). If I were to remove the on-page-change attribute from one of the pagination controls then it all works correctly. However I prefer not to have to do that because in my real life case I use different pagination templates for different screen sizes and my code is more like this plunk (albeit with the pagination controls wrapped up in my own directive)

[http://plnkr.co/edit/wbNOx6OLo79GPgU3CUpf?p=info]

Now we get at least 4 hits to the API each time I change page...see the console log

I hope you understand my issue and can point me in the right direction

Many thanks in advance

**Comment by michaelbromley** [comment]
Hi,

Thanks for the clear demos. I have a few suggestions:

1. If you were to use `ng-if` on your controls directives, the extra callbacks would not fire. In your case this might not be desirable since you are reacting to screen size, and tracking that in JS may be too much of a burden.
2. You can safely remove all but one of the `on-page-change` attributes and, assuming the one *with* the attribute set is in your view, it will fire when *any* of the linked controls are clicked.
3. The best way to go might be to get rid of that callback altogether and instead just use a `$scope.$watch` on the value of `currentPage`. This means that it does not matter how you change the page - it will only invoke your page change function once. I modified your last demo to show this: http://plnkr.co/edit/lM8UKlPwSWoiFB4C364U?p=preview

**Comment by hplloyd** [comment]
I think option 3 is the way ahead... many thanks I can make that work :)

**Comment by hplloyd** [comment]
Alternatively, I might check that the page has changed by storing the last page requested in a variable and block calls that are not genuinely for a new page

**Comment by MangeshPimpalkar** [comment]
+1

**Comment by michaelbromley** [comment]
@hplloyd Yes, that's a good suggestion. I'm going to leave this issue open with an "enhancement" label to remind myself to consider implementing when I make the next update to the lib (no idea when that will be at present).

**Comment by hplloyd** [comment]
No worries, my fix is working fine for me until you implement :)

**Comment by GregaVrbancic** [comment]
Any news regarding this issue? Or is it still the easiest way to solve this issue the way @hplloyd solved it?

**Comment by michaelbromley** [comment]
@GregaVrbancic For now I have made not incorporated a fix into the code. Just go with one of the work-arounds mentioned above. To be honest, this module is not something I require for my work projects, and is therefore not high on my list of priorities for spending time on.
